### PR TITLE
Gragh

### PR DESCRIPTION
**Pull request by xialiu1988**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: Data-Structures/src/main/java/Graph/Graph.java [hunk 1]
```diff
@@ -1,4 +1,191 @@
 package Graph;
 
+import java.util.*;
+
 public class Graph {
+
+    public List<Vertex> Vertices;
```
Comment (evanslaton) [inline Data-Structures/src/main/java/Graph/Graph.java]: Variable name should start with a lowercase letter -> `vertices`

File: Data-Structures/src/main/java/Graph/Graph.java [hunk 2]
```diff
@@ -1,4 +1,191 @@
...
+            return null;
+        }
+        else
+        {
+            Vertices.add(node);
+            return node;
+        }
+    }
+
+    //addedge()Adds a new edge between two nodes in the graph
```
Comment (evanslaton) [inline Data-Structures/src/main/java/Graph/Graph.java]: Corpse code!

File: Data-Structures/src/test/java/Graph/GraphTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        Vertex v2 = g.addNode("b");
+        Vertex v3 = g.addNode("c");
+        g.addEdge(v1, v2, 10);
+        g.addEdge(v1, v3, 12);
+        List<Vertex> list=g.getNeighbors(v1);
+        assertTrue(list.size() == 2);
+    }
+
+    @Test
+    public void getNeighborswithweghts() {
```
Comment (evanslaton) [inline Data-Structures/src/test/java/Graph/GraphTest.java]: Remember to use camelCase!

File: Data-Structures/src/main/java/Graph/Graph.java (hunk 1 above)
Comment (xialiu1988) [reply Data-Structures/src/main/java/Graph/Graph.java]: yeah forgot to change that...
